cargo-cyclonedx/src/generator.rs

```diff
@@ -110,232 +115,242 @@ impl SbomGenerator {
-            Err(e) => log::error!(
-                "Package {} has an invalid repository URI ({}): {} ",
-                package.name(),
-                vcs,
-                e
-            ),
+        if let Some(vcs) = &metadata.repository {
+            match Uri::try_from(vcs.to_string()) {
+                Ok(uri) => references.push(ExternalReference::new(ExternalReferenceType::Vcs, uri)),
+                Err(e) => log::error!(
+                    "Package {} has an invalid repository URI ({}): {} ",
+                    package.name(),
+                    vcs,
+                    e
+                ),
+            }
         }
-    }
 
-    if !references.is_empty() {
-        return Some(ExternalReferences(references));
+        if !references.is_empty() {
+            return Some(ExternalReferences(references));
+        }
+
+        None
     }
 
-    None
-}
+    fn get_licenses(&self, package: &Package) -> Option<Licenses> {
+        let mut licenses = vec![];
 
-fn get_licenses(package: &Package) -> Option<Licenses> {
-    let mut licenses = vec![];
+        if let Some(license) = package.manifest().metadata().license.as_ref() {
+            let strict = match &self.config.license_parser {
+                Some(opts) => opts.strict,
+                None => false,
+            };
+            let result = match strict {
+                true => SpdxExpression::try_from(license.to_string()),
+                false => SpdxExpression::parse_lax(license.to_string()),
+            };
 
-    if let Some(license) = package.manifest().metadata().license.as_ref() {
-        match SpdxExpression::try_from(license.to_string()) {
-            Ok(expression) => licenses.push(LicenseChoice::Expression(expression)),
-            Err(err) => {
-                log::error!(
-                    "Package {} has an invalid license expression, trying lax parsing ({}): {}",
-                    package.name(),
-                    license,
-                    err
-                );
-
-                match SpdxExpression::parse_lax(license.to_string()) {
-                    Ok(expression) => licenses.push(LicenseChoice::Expression(expression)),
-                    Err(err) => {
-                        log::error!(
-                        "Package {} has an invalid license expression that could not be converted to a valid expression, using named license ({}): {}",
+            match result {
+                Ok(expression) => licenses.push(LicenseChoice::Expression(expression)),
+                Err(err) => {
+                    let level = match &self.config.license_parser {
+                        Some(opts) if opts.accept_named.contains(license) => Level::Info,
+                        _ => Level::Warn,
+                    };
+                    log::log!(
+                        level,
+                        "Package {} has an invalid license expression (error: {}, strict mode: {}), using named license: {}",
```

Comment:
We'll want to make this error message consistent with what we use for CLI names. I'd suggest we also put the error message at the end and not include the strict mode bit (maybe have a second log message at `debug` level).